src/lib/keys/KeyManager.ts

```diff
@@ -519,17 +522,19 @@ class KeyManager {
       }
     }
     // Read in file buffer and signature
-    const fileBuffer = this.fileSystem.readFileSync(filePath);
-    const isVerified = await this.verifyData(fileBuffer, keyBuffer!);
-    return isVerified;
+    const signatureBuffer = this.fileSystem.readFileSync(filePath);
+    const verifiedMessage = await this.verifyData(signatureBuffer, keyBuffer!);
+    this.fileSystem.writeFileSync(filePath, verifiedMessage);
+
+    return true;
   }
 
   /**
    * Encrypts the given data for a specific public key
    * @param data The data to be encrypted
    * @param publicKey The key to encrypt for
    */
-  async encryptData(data: Buffer, publicKey?: Buffer): Promise<string> {
+  async encryptData(data: Buffer, publicKey?: Buffer): Promise<Buffer> {
```

Comment:
Shouldn't encryption and decryption be handled directly by a proper cipher model? The keymanager is only meant to manage keys. That's all.

Reply:
yeah we can probably put them somewhere else, perhaps in our vault model? but I just thought since we don't have any specific `cryptography` domain yet they should just be closest to the keys.